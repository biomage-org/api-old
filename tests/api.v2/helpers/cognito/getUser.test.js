
jest.mock('../../../../src/api.v2/helpers/cognito/getAwsPoolId');


jest.mock('../../../../src/config', () => ({
  cognitoISP: {
    adminGetUser: jest.fn(({ Username }) => ({
      promise: () => Promise.resolve({
        UserAttributes: [
          { Name: 'name', Value: `${Username}-test` },
          { Name: 'email', Value: `${Username}@example.com` },
        ],
      }),
    })),
  },
}));


const getUser = require('../../../../src/api.v2/helpers/cognito/getUser');

describe('getUser', () => {
  it('returns a valid user', async () => {
    const userInfo = await getUser('email');
    expect(userInfo).toMatchSnapshot();
  });
});
