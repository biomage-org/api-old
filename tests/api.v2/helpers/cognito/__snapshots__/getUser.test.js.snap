// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUser returns a valid user 1`] = `
Array [
  Object {
    "Name": "name",
    "Value": "email-test",
  },
  Object {
    "Name": "email",
    "Value": "email@example.com",
  },
]
`;
